### DISCUSSION TITLE
Is there any way to compare the new data with the previous data with useQuery

### DISCUSSION THREAD

**Discussion by lukesmurray** [TanStack/query#1890] [Q&A] [answered]
I have a dictionary of booleans I request which lets me know if some rather expensive workflows have been updated.

```
{
workflowADone: true,
workflowBDone: false,
workflowCDone: true,
}
```

I use a query to fetch  this dictionary but I only want to perform an action if one of these flags has changed from the previous time I've called `useQuery`.
Is there a way for me to always access both the `previousData` and the `currentData` while using `useQuery`?

**Comment by TkDodo** [comment, marked as answer]
no, it's not supported currently. It is usually enough to depend on referential identity when running an effect or so. If your list of flags is deterministic, you can do:

```
const { data } = useMyWorkflowQuery()
const { workflowADone, workflowBDone } = data


const aData = React.useMemo(() => worflowADone ? expensiveComputation() : null, [worflowADone, expensiveComputation])
const bData = React.useMemo(() => worflowBDone ? expensiveComputation() : null, [worflowBDone, expensiveComputation])

```

Otherwise, you'd need to keep track of the previous one yourselves. There is the [usePrevious](https://usehooks.com/usePrevious/) hook which does that via a ref, or, you can just drag the previous result along in your fetchFn, something like:

```
type ResultWithPrevious<T> = {
  current: T
  previous?: T
}
const fetchWithPrevious = async <T>(url: string, previous?: ResultWithPrevious<T>): Promise<ResultWithPrevious<T> => {
  const { data } = await axios.get(url)
  
  return { previous: previous.current, current: data }
}

const useTodos = () => {
  const queryClient = useQueryClient()
  const previousTodos = queryClient.getQueryData<Todos>('todos')
  return useQuery('todos', () => fetchWithPrevious<Todos>('/todos', previousTodos))
}
```

```
const { data } = useTodos()
data.current // Todos
data.previous // Todos | undefined
```

I realise that this is a bit of boilerplate, but it basically means getting the data from the cache before doing the "next" fetch and keeping it around.

**Reply by lukesmurray** [reply]
thanks for clearing that up for me and providing a couple of work arounds!
